botocore/signers.py

```diff
@@ -123,6 +113,36 @@ def sign(self, operation_name, request):
                                             signature_version)
             signer.add_auth(request=request)
 
+    def choose_signer(self, operation_name, suffix=''):
+        """
+        Allow overriding the signature version via the choose-signer event.
+        A value of `botocore.UNSIGNED` means no signing will be performed.
+
+        :param operation_name: The operation to sign.
+        :param suffix: A suffix that needs to be on all valid signers. This
```

Comment:
Public in this context as being called as an API within our botocore code (non `_internal`).  I think there's an increasingly higher bar for sensible APIs going from `_internal` methods used within a class, non-internal methods used amongst botocore classes, and methods exposed for customers to call.

I made another comment below about an alternate approach you could use.